src/entries/transfer.rs

```diff
@@ -27,7 +29,7 @@ pub fn generate_transfer_smt(transfer_req: TransferReq) -> Result<(String, Strin
             .collect(),
     );
     let sender_withdrawals = get_withdrawal_cota_by_lock_hash(
-        transfer_req.withdrawal_lock_hash,
+        blake2b_256(transfer_req.withdrawal_lock_script.clone()),
```

Comment:
Unnecessary clone, `blake2b_256` accepts generic type `AsRef<[u8]>`
```suggestion
        blake2b_256(transfer_req.withdrawal_lock_script.as_slice()),
```